File: NewsAnalysis/explore_db.js
import pkg from "pg";
const { Pool } = pkg;
import dotenv from "dotenv";

dotenv.config({ path: '../Stockify-Backend/.env' });

const db = new Pool({
  connectionString: process.env.DATABASE_URL,
  ssl: {
    rejectUnauthorized: false
  }
});

async function check() {
  try {
    const res = await db.query(`
        SELECT table_name
        FROM information_schema.tables
        WHERE table_schema = 'public'
    `);
    console.log("Tables:", res.rows.map(r => r.table_name));
    for (const row of res.rows) {
        const colRes = await db.query(`SELECT column_name, data_type FROM information_schema.columns WHERE table_name = $1`, [row.table_name]);
        console.log(`Table ${row.table_name}:`, colRes.rows);
    }
  } catch (e) {
    console.error(e);
  } finally {
    await db.end();
  }
}

check();

File: NewsAnalysis/explore_db.py
import os
import psycopg2
from dotenv import load_dotenv

load_dotenv('../Stockify-Backend/.env')

DATABASE_URL = os.getenv('DATABASE_URL')
print('DATABASE_URL:', DATABASE_URL)
try:
    conn = psycopg2.connect(DATABASE_URL)
    cur = conn.cursor()
    cur.execute('''
        SELECT table_name
        FROM information_schema.tables
        WHERE table_schema = 'public'
    ''')
    tables = cur.fetchall()
    print('Tables:', tables)
    for t in tables:
        cur.execute(f"SELECT column_name, data_type FROM information_schema.columns WHERE table_name = '{t[0]}'")
        print(f"Table {t[0]}: {cur.fetchall()}")
except Exception as e:
    print('Error:', e)
